src/raftstore/store/store.rs

```diff
@@ -302,17 +339,28 @@ impl<T, C> Store<T, C> {
         Ok(())
     }
 
-    fn clear_stale_meta(&mut self, wb: &mut WriteBatch, region: &metapb::Region) {
+    fn clear_stale_meta(
+        &mut self,
+        kv_wb: &mut WriteBatch,
+        raft_wb: &mut WriteBatch,
+        region: &metapb::Region,
+    ) {
         let raft_key = keys::raft_state_key(region.get_id());
-        let raft_state: RaftLocalState =
-            match self.engine.get_msg_cf(CF_RAFT, &raft_key).unwrap() {
-                // it has been cleaned up.
-                None => return,
-                Some(value) => value,
-            };
+        let raft_state = match self.raft_engine.get_msg(&raft_key).unwrap() {
```

Comment:
as we can't destroy a peer which is applying snapshot, we don't need check snapshot_raft_state